lib/spell-check-view.coffee

```diff
@@ -107,3 +110,64 @@ class SpellCheckView
     instance = @getInstance()
     misspelling = @editor.getTextInBufferRange marker.getBufferRange()
     instance.suggest args, misspelling
+
+  addContextMenuEntries: (mouseEvent) =>
+    @clearContextMenuEntries()
+    # Get buffer position of the right click event. If the click happens outside
+    # the boundaries of any text, the method defaults to the buffer position of
+    # the last character in the editor.
+    currentScreenPosition = atom.views.getView(@editor).component.screenPositionForMouseEvent mouseEvent
+    currentBufferPosition = @editor.bufferPositionForScreenPosition(currentScreenPosition)
+
+    # Check to see if the selected word is incorrect.
+    if marker = @markerLayer.findMarkers({containsBufferPosition: currentBufferPosition})[0]
+      corrections = @getCorrections(marker)
+      if corrections.length > 0
+        @spellCheckModule.contextMenuEntries.push({
+          menuItem: atom.contextMenu.add({'atom-text-editor': [{type: 'separator'}]})
+        })
+
+        for correction in corrections
+          contextMenuEntry = {}
+          # Register new command for correction.
+          contextMenuEntry.command = do (correction, contextMenuEntry) =>
+            atom.commands.add atom.views.getView(@editor),
+              'spell-check:correct-misspelling-' + correction.index, =>
+                @makeCorrection(correction, marker)
+                @clearContextMenuEntries()
+
+          # Add new menu item for correction.
+          contextMenuEntry.menuItem = atom.contextMenu.add({
+            'atom-text-editor': [{label: correction.label, command: 'spell-check:correct-misspelling-' + correction.index}]
+          })
+          @spellCheckModule.contextMenuEntries.push contextMenuEntry
+
+  makeCorrection: (correction, marker) =>
+    if correction.isSuggestion
+      # Update the buffer with the correction.
+      @editor.setSelectedBufferRange(marker.getBufferRange())
+      @editor.insertText(correction.suggestion)
+    else
```

Comment:
I had brought this use case over from the `CorrectionsView:confirm()` function to preserve the 'Add to Known Words' functionality, it looks like that's the only way it's being used right now. I added a spec now to test adding a known word from the context menu.